### ISSUE TITLE
Broken support for fragmented TTML subtitles

### ISSUE THREAD

**Issue by stepashka69** [Orange-OpenSource/hasplayer.js#126]
At hasplayer v 1.4.0 the support for TTML subtitles split to fragments is broken (see sample manifest file attached). 

This happens because of changes introduced in commit a3012ae99a40a86880a7129003fc8e192b13d929 .

The problem is that at `app/js/mss/MssFragmentController.js:393` the request result for single complete XML file returns combination of the same XMLs. When I tried to copy it from browser it was around 15 Mb of text (I've attached a part of it). This cause an error further at `app/js/streaming/captioning/TextTTMLXMLMP4SourceBuffer.js:220` . XML fails to parse (because it is a concatenation of many XMLs) and subtitles are not added to a video...

Manifest file is similar as in https://github.com/Orange-OpenSource/hasplayer.js/issues/123  .

Sorry, I'm not familiar about your player's logic at that part of the code so I can not offer a solution. Feel free to ask more questions about what I found during my debug. 

[manifest.xml.zip](https://github.com/Orange-OpenSource/hasplayer.js/files/442482/manifest.xml.zip)
[correct.xml.zip](https://github.com/Orange-OpenSource/hasplayer.js/files/442477/correct.xml.zip)
[concatxml.xml.zip](https://github.com/Orange-OpenSource/hasplayer.js/files/442495/concatxml.xml.zip)

**Comment by nicosang** [comment]
I don't really understand your issue. could you, please, give me directly the url of your stream? I will test it by myself.

Thanks,

**Comment by stepashka69** [comment]
Let me try again...

As you can see from the manifest file attached the subtitles are split to a fragments. When video is playing the browser sends the following requests:

```
http://iis.net/CCTEST03.ism/QualityLevels(48000)/Fragments(caption_1=5730000000)
http://iis.net/CCTEST03.ism/QualityLevels(48000)/Fragments(caption_1=5760000000)
http://iis.net/CCTEST03.ism/QualityLevels(48000)/Fragments(caption_1=5790000000)
```

and so on...

Each response has something like this

```
<?xml version="1.0" encoding="UTF-8"?><tt xml:lang="eng" xmlns="http://www.w3.org/ns/ttml" xmlns:tts="http://www.w3.org/ns/ttml#styling" xmlns:smpte="http://www.smpte-ra.org/schemas/2052-1/2010/smpte-tt" ><head>
<smpte:information smpte:mode="Enhanced" />
<styling>
<style xml:id="emb" tts:fontSize="4.1%" tts:fontFamily="monospaceSansSerif" />
<style xml:id="ttx" tts:fontSize="3.21%" tts:fontFamily="monospaceSansSerif"/>
<style xml:id="backgroundStyle" tts:fontFamily="proportionalSansSerif" tts:fontSize="18px" tts:textAlign="center" tts:origin="0% 66%" tts:extent="100% 33%" tts:backgroundColor="rgba(0,0,0,0)" tts:displayAlign="center" />
<style xml:id="speakerStyle" style="backgroundStyle" tts:color="white" tts:textOutline="black 1px" tts:backgroundColor="transparent" />
<style xml:id="textStyle" style="speakerStyle" tts:color="white" tts:textOutline="none" tts:backgroundColor="black" />
</styling><layout>
<region xml:id="full" tts:origin="0% 0%" tts:extent="100% 100%" tts:zIndex="1" />
<region xml:id="speaker" style="speakerStyle" tts:zIndex="1" />
<region xml:id="background" style="backgroundStyle" tts:zIndex="0" />
</layout>
</head><body><div>

<!-- text-based TTML -->
	<p begin="00:00:00.000" end="00:00:02.960" region="speaker"><span style="textStyle">some of them for 20 years or longer.</span></p>
</div></body></tt>
```

The problem is that after the changes done at the commit I've mentioned above all these files got combined in one big file (including `<?xml versi...` ) which makes XML incorrect. Then there is an attempt of parsing all that combined data which obviously fails. 

All the details where it happens (for 1.4.0 release) is in my previous message.

For one xml file it works ok. It fails only when subtitles are fragmented. Probably it is not the most common way of doing it but unfortunately it is a standard way of preparing data for our customers.

PS: Unfortunately I can not give you the access to the video files. If you have further questions - feel free to ask.

**Comment by nicosang** [comment]
I've tested the 'small' xml file. Indeed, the file is well formed. I don't understand why the files are combined together. I'm not able to reproduce it with any of my streams. Could you show me debug informations from hasplayer, please?
